test/test_sessionmanager.py

```diff
@@ -0,0 +1,105 @@
+import unittest
+
+from urllib3.sessionmanager import SessionManager
+from urllib3.poolmanager import ProxyManager, PoolManager
+from urllib3.util.sessioncontext import DefaultCookiePolicy, CookieJar, SessionContext
+from urllib3.packages import six
+from urllib3.response import HTTPResponse
+from urllib3.request import Request
+
+cookielib = six.moves.http_cookiejar
+
+
+class TestSessionManager(unittest.TestCase):
+
+    def setUp(self):
+        self.manager = SessionManager(PoolManager())
+
+    def test_set_cookie_policy(self):
+        policy = cookielib.DefaultCookiePolicy()
+        self.assertNotEqual(policy, self.manager.context.cookie_jar._policy)
+        self.manager.context.cookie_jar.set_policy(policy)
```

Comment:
So this seems *really* clunky: we're well past a Law of Demeter violation here. Perhaps the `SessionManager` should expose helpers for things like this, rather than expect users to patch this many objects deep?